uvicorn/protocols/http/h11_impl.py

```diff
@@ -367,6 +379,11 @@ def __init__(
         self.response_started = False
         self.response_complete = False
 
+        # Sendfile
+        if self.allow_sendfile:  # pragma: no cover
+            # Set the buffer to 0 to avoid the problem of sending file before headers.
+            transport.set_write_buffer_limits(0)
```

Comment:
I'm not really comfortable with the fact that this'll change the behaviour throughout, regardless of if `http.response.zerocopysend` is actually being used or not for any given request/response.

Reply:
If this option is not configured, the headers may be sent after the body is sent.